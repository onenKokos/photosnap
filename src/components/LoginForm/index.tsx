import React from 'react';
import { useState, ChangeEvent, FormEvent } from 'react';

export interface LoginFormProps {
  shouldRemember: boolean;
  onUsernameChange: (username: string) => void;
  onPasswordChange: (password: string) => void;
  onRememberChange: (remember: boolean) => void;
  onSubmit: (username: string, password: string, remember: boolean) => void;
}

function LoginForm({
  shouldRemember,
  onUsernameChange,
  onPasswordChange,
  onRememberChange,
  onSubmit,
}: LoginFormProps) {
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');
  const [remember, setRemember] = React.useState(shouldRemember);

  const handleUsernameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    setUsername(value);
    onUsernameChange(value);
  };

  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    setPassword(value);
    onPasswordChange(value);
  };

  const handleRememberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { checked } = e.target;
    setRemember(checked);
    onRememberChange(checked);
  };

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    onSubmit(username, password, remember);
  };

  return (
    <form data-testid="login-form" onSubmit={handleSubmit}>
      <label htmlFor="username">Username:</label>
      <input
        data-testid="username"
        type="text"
        name="username"
        value={username}
        onChange={handleUsernameChange}
      />

      <label htmlFor="password">Password:</label>
      <input
        data-testid="password"
        type="password"
        name="password"
        value={password}
        onChange={handlePasswordChange}
      />

      <label>
        <input
          data-testid="remember"
          name="remember"
          type="checkbox"
          checked={remember}
          onChange={handleRememberChange}
        />
        Remember me?
      </label>

      <button type="submit" data-testid="submit">
        Sign in
      </button>
    </form>
  );
}

export default LoginForm;
