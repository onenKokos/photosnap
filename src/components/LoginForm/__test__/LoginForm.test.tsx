import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import LoginForm, { LoginFormProps } from '../';

const renderLoginForm = (props: Partial<LoginFormProps> = {}) => {
  const defaultProps: LoginFormProps = {
    onPasswordChange() {
      return;
    },
    onRememberChange() {
      return;
    },
    onUsernameChange() {
      return;
    },
    onSubmit() {
      return;
    },
    shouldRemember: true,
  };
  return render(<LoginForm {...defaultProps} {...props} />);
};

describe('<LoginForm />', () => {
  test('should display a blank login form, with remember me checked by default', async () => {
    const { findByTestId } = renderLoginForm();

    const loginForm = await findByTestId('login-form');

    expect(loginForm).toHaveFormValues({
      username: '',
      password: '',
      remember: true,
    });
  });
});
