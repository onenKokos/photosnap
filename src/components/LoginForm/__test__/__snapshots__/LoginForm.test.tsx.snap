// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: LoginForm should render correctly 1`] = `
<div>
  <form
    data-testid="login-form"
  >
    <label
      for="username"
    >
      Username:
    </label>
    <input
      data-testid="username"
      name="username"
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password:
    </label>
    <input
      data-testid="password"
      name="password"
      type="password"
      value=""
    />
    <label>
      <input
        checked=""
        data-testid="remember"
        name="remember"
        type="checkbox"
      />
      Remember me?
    </label>
    <button
      data-testid="submit"
      type="submit"
    >
      Sign in
    </button>
  </form>
</div>
`;
